### ISSUE TITLE
Current measurements via `get_current` are off by factor 4 under specific circumstances

### ISSUE THREAD

**Issue by redfast00** [GSI-HPC/py-ina238#10]
If you first do `set_shunt_calibration`, then `set_config(adc_range=ti.ina238.ADCRange.LOW)`, the `SHUNT_CAL` register is not updated. This causes the `get_current` function to return results that are off by a factor 4 (https://github.com/GSI-HPC/py-ina238/blob/main/src/ti/ina238.py#L278).

As a workaround, you can call `set_config(adc_range=ti.ina238.ADCRange.LOW)` first, then after that call `set_shunt_calibration`. Alternatively, you can use `get_shunt_voltage` and do the current calculation yourself in Python.

**Comment by dennisklein** [comment]
Unfortunately, I have not yet found the time to add proper documentation nor unit testing (as well as support for the `DIAG_ALRT` register). But here is what the documentation should have provided: The existing API (so far) is designed as a very low-level and eager abstraction which merely adds some pythonic niceness on top of the raw device. So, no dependencies between registers such as you describe are really modeled yet. As a consequence the user has to be very familiar with the device and do things in the right order.

However, I was debating already with myself how a higher level API could/should look like but had no progress due to lack of time/priority. The eagerness of the current API also leaves out opportunities to skip any device communication which is effectively a no-op. On the other hand, it should give the most control which was the first concern for our internal use case.

I would like to leave this open to be sure to address this once documentation is produced - thank you!

**Comment by redfast00** [comment]
I just encountered another issue:

```
self.inar = ti.ina238.Driver(I2CDevice(self.i2c, 0x43))
self.inal = ti.ina238.Driver(I2CDevice(self.i2c, 0x47))
self.inal.reset()
self.inar.reset()
self.inal.set_config(adc_range=ti.ina238.ADCRange.LOW)
self.inar.set_config(adc_range=ti.ina238.ADCRange.LOW)
self.inal.set_shunt_calibration(1.0, 0.04) # 1 ohm, max 40mA
self.inar.set_shunt_calibration(1.0, 0.04) # 1 ohm, max 40mA
self.inal.set_adc_config(avg_count=ti.ina238.Samples.AVG_64, vshunt=ti.ina238.ConversionTime.T_4120_US, mode=ti.ina238.Mode.CONTINUOUS_VBUS_VSHUNT_DIETEMP)
self.inar.set_adc_config(avg_count=ti.ina238.Samples.AVG_64, vshunt=ti.ina238.ConversionTime.T_4120_US, mode=ti.ina238.Mode.CONTINUOUS_VBUS_VSHUNT_DIETEMP)
```

This does not work as expected (this time, a factor of 4 difference between shunt voltage and current, but in the opposite direction as the previous issue), but if you add:

```
self.inal.get_config()
self.inar.get_config()
```

at the end, it does work. Getters should not impact internal state, but here they do (https://github.com/GSI-HPC/py-ina238/blob/main/src/ti/ina238.py#L210)

**Comment by dennisklein** [comment]
I agree, would you like to provide a PR that moves [this logic](https://github.com/GSI-HPC/py-ina238/blob/d24727ab4eb498402a0a19ffb9fc62090e03ef37/src/ti/ina238.py#L209-L212) to the setter? Or, if you have something different in mind, let's discuss a fix.
